DataFormats/L1Trigger/interface/RegionalOutput.h

```diff
@@ -123,15 +123,15 @@ namespace l1t {
     const_iterator begin() const { return const_iterator(this, 0); }
     const_iterator end() const { return const_iterator(this, values_.size()); }
 
-    Region region(unsigned int ireg) const { return Region(this, ireg == 0 ? 0 : regions_[ireg - 1], regions_[ireg]); }
+    Region region(unsigned int ireg) const { return Region(this, (ireg == 0 or ireg>=regions_.size()) ? 0 : regions_[ireg - 1], regions_[ireg]); }
```

Comment:
Wait, I think this fix makes it worse than before.
Before, ireg >=regions_.size() would have given random data or crash, now it silently creates a region spannig the whole detector.
I think the fix Adriano was asking for is more along the lines of
```
Region region(unsigned int ireg) const { 
     if (ireg >= regions_.size()) throw cms::Exception("Region index out of bounds");
     return Region(this, ireg == 0 ? 0 : regions_[ireg - 1], regions_[ireg]); 
}
```

Reply:
Yes indeed, something like this that would make clear, eventually, why the segfault/crash is happening.

Reply:
Done